### ISSUE TITLE
`unneeded_escaping` false positive in 0.63.0-rc.2 (autoclosure forwarding)

### ISSUE THREAD

**Issue by shagedorn** [realm/SwiftLint#6391]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

When writing a function that accepts an autoclosure which it passes on to another function that requires an escaping autoclosure, SwiftLint falsely claims that `@escaping` can be omitted at the outer function:

```
class MyClass {
    private var storedClosure: (() -> String)?

    func outer(closure: @escaping @autoclosure () -> String) {
        inner(closure: closure())
    }
    func inner(closure: @escaping @autoclosure () -> String) {
        storedClosure = closure
    }
}
```

Linting this code throws an `unneeded_escaping` warning in the outer function's signature, but if you remove it, the compiler throws an error:

> Escaping autoclosure captures non-escaping parameter 'closure'

This is not limited to functions within the same type.

### Environment

SwiftLint version: 0.63.0-rc.2
Xcode version: 26.2
Installation method used: release ZIP file download
Configuration file: irrelevant, except that `unneeded_escaping` is enabled

**Comment by shagedorn** [comment]
Thank you ❤️
